can_tools/scrapers/official/base.py

```diff
@@ -120,6 +120,12 @@ def _put_exec(self, engine: Engine, df: pd.DataFrame) -> None:
                 sess.commit()
                 rows_deleted = res_delete.rowcount
                 print("Removed the {} rows from temp table".format(rows_deleted))
+            
+        if rows_inserted == 0:
+            raise InsertedZeroRowsError(
+                "_put_exec() method completed successfully "
+                "but inserted zero rows into database."
+            )
 
         return worked, rows_inserted, rows_deleted
```

Comment:
We have this `worked` variable that is supposed to return `True` or `False` depending on the outcome of this method. We [don't ever do anything with it](https://github.com/covid-projections/can-scrapers/blob/main/services/prefect/flows/generated_flows.py#L66-L67), and I think it will only return ever return `True` because (AFAICT) if an exception is thrown the method will terminate. 

So, I found it easiest just to raise an exception in this method directly

Reply:
Looks like the code above uses `fast_append_to_sql()` to write the data from the dataframe into a temporary table and then the actual insert is copying the data over from there.  So I wonder if there's an (easy) way to make `fast_append_to_sql()` log the number of rows it inserted into the temporary table (or anything else that'd be useful).  That way if this happens again we'll which part failed. 🤷 

Food for thought. Doesn't need to be in this PR.

Reply:
agreed, that would give us some more insight. I think I'm going to ship this as is, but can add that to the backlog